src/utils/daemon.js

```diff
@@ -1,7 +1,12 @@
 import IPFSFactory from 'ipfsd-ctl'
 import logger from './logger'
+import { join } from 'path'
+import { spawnSync } from 'child_process'
+import { dialog } from 'electron'
```

Comment:
This file now mixes multiple concerns. Let's split out the external process finding and calling and the electron dialog elements into small functions that we can test and compose.